### ISSUE TITLE
for 24hr/12hr formatting for timestamps?

### ISSUE THREAD

**Issue by hardcodet** [date-fns/date-fns#1129]
- date-fns latest (alpha-27)

TL/DR:
Can we specify time formatting with `formatRelative`? If not, would it be feasible to add this as an explicit parameter?

Context:
`formatRelative` is awesome, but doesn't take into account that languages and time formatting not necessarily match. For example, I am Swiss (24 hour timestamps), but my phone/tablet/... are all set to US English. Accordingly, the function resolves to something like `today at 7:30 PM`, when my phone would show `19:30` for the that time. So as a user, I would expect `today 19:30`. From a UX point of view, this is a bit of a show-stopper.

So I would love to see smth like this:

    formatRelative (timestamp, new Date(), {locale: en, timeFormat: "24hours"});

Apart from that, awesome work - thank you so much for this! 🚀

**Comment by dkozickis** [comment]
It feels like something that has to be controlled by application that is using `date-fns`. As in - application must be able to provide correct locale to `formatRelative` 3rd argument.

**Comment by Jav3k** [comment]
Example of one of solution with custom formatRelative function in locale object provided here: https://github.com/date-fns/date-fns/issues/1218#issuecomment-599182307
